homeassistant/components/waze_travel_time/__init__.py

```diff
@@ -1,24 +1,188 @@
 """The waze_travel_time component."""
 
 import asyncio
+import logging
+
+from pywaze.route_calculator import CalcRoutesResponse, WazeRouteCalculator, WRCError
+import voluptuous as vol
 
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import Platform
-from homeassistant.core import HomeAssistant
+from homeassistant.const import CONF_REGION, Platform
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+)
+from homeassistant.helpers.httpx_client import get_async_client
+from homeassistant.helpers.selector import (
+    BooleanSelector,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+    TextSelector,
+)
 
-from .const import DOMAIN, SEMAPHORE
+from .const import (
+    CONF_AVOID_FERRIES,
+    CONF_AVOID_SUBSCRIPTION_ROADS,
+    CONF_AVOID_TOLL_ROADS,
+    CONF_DESTINATION,
+    CONF_EXCL_FILTER,
+    CONF_INCL_FILTER,
+    CONF_ORIGIN,
+    CONF_REALTIME,
+    CONF_UNITS,
+    CONF_VEHICLE_TYPE,
+    DEFAULT_VEHICLE_TYPE,
+    DOMAIN,
+    METRIC_UNITS,
+    REGIONS,
+    SEMAPHORE,
+    UNITS,
+    VEHICLE_TYPES,
+)
 
 PLATFORMS = [Platform.SENSOR]
 
+SERVICE_GET_TRAVEL_TIMES = "get_travel_times"
+SERVICE_GET_TRAVEL_TIMES_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ORIGIN): TextSelector(),
+        vol.Required(CONF_DESTINATION): TextSelector(),
+        vol.Required(CONF_REGION): SelectSelector(
+            SelectSelectorConfig(
+                options=sorted(REGIONS),
+                mode=SelectSelectorMode.DROPDOWN,
+                translation_key=CONF_REGION,
+            )
+        ),
+        vol.Optional(CONF_INCL_FILTER, default=""): TextSelector(),
+        vol.Optional(CONF_EXCL_FILTER, default=""): TextSelector(),
```

Comment:
Please remove these filters since the backing API does not support it.

Reply:
Good idea. Does not make sense if this service should return all possible routes